c7n/resources/account.py

```diff
@@ -424,6 +425,66 @@ def process(self, resources, event=None):
         return []
 
 
+@actions.register('set-password-policy')
+class SetAccountPasswordPolicy(BaseAction):
+    """Set an account's password policy.
+
+    This only changes the policy for the items provided.
+    If this is the first time setting a password policy and an item is not provided it will be
+    set to the defaults defined in the boto docs for IAM.Client.update_account_password_policy
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: set-account-password-policy
+                resource: account
+                filters:
+                  - not:
+                    - type: password-policy
+                      key: MinimumPasswordLength
+                      value: 10
+                      op: ge
+                actions:
+                    - type: set-password-policy
+                      policy:
+                        MinimumPasswordLength: 20
+    """
+    schema = type_schema(
+        'set-password-policy',
+        policy={
+            'type': 'object'
+        })
+    shape = 'UpdateAccountPasswordPolicyRequest'
+    service = 'iam'
+    permissions = ('iam:GetAccountPasswordPolicy', 'iam:UpdateAccountPasswordPolicy')
+
+    def validate(self):
+        return shape_validate(
+            self.data.get('policy', {}),
+            self.shape,
+            self.service)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('iam')
+        account = resources[0]
+        if account.get('c7n:password_policy'):
+            config = account['c7n:password_policy']
+        else:
+            try:
+                config = client.get_account_password_policy().get('PasswordPolicy')
+            except client.exceptions.NoSuchEntityException:
+                config = {}
+        for k, v in self.data.get('policy').items():
```

Comment:
for a shallow copy just do dict(self.data['policy'])

Reply:
```python
params = dict(self.data['policy'])
params.update(config)
```

Reply:
this does read better. I rebased locally and have some merge conflicts to adjust. Out of the house right now but will push up again when back.